bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -770,8 +781,13 @@ uint64_t BCStateTran::uniqueMsgSeqNum() {
     lastMilliOfUniqueFetchID_ = milli;
     lastCountOfUniqueFetchID_ = 0;
   } else {
-    if (lastCountOfUniqueFetchID_ == 0x3FFFFF) lastMilliOfUniqueFetchID_++;
-    lastCountOfUniqueFetchID_++;
+    if (lastCountOfUniqueFetchID_ == 0x3FFFFF) {
+      LOG_WARN(STLogger, "BCStateTran::uniqueMsg: SeqNum Counter reached max value");
+      lastMilliOfUniqueFetchID_++;
+      lastCountOfUniqueFetchID_=0;

```

Comment:
This fix should be verified via a test.

Reply:
It's an overflow that was already fixed and discussed a while back. See https://github.com/vmware/concord-bft/pull/78. I just copied over the fix. I could add a test for this but I don't think it's priority.

The right thing to do is to put logic like this into a separate library with tests and use it from multiple places rather than duplicating.